build: Pin jupyter client version

ddn0 opened the pull request:
The pytest-notebook package depends on a older version of nbconvert
which in turn uses a nbclient package that seems incompatible with the
recent jupyter-client 7.

This leads to errors like:

  TypeError: 'coroutine' object is not subscriptable

tylershunt reviewed (approved):
lgtm

tylershunt commented:
> The idea I'll try to socialize here is that just like we spent some effort on optimizing graph load, we might need to put in some effort on graph write (or transaction commit). This would entail restructuring our graph writing code to overlap IO with computation. As we gain more experience with transactions on "real" graphs the performance traps should become more clear. But the current structure is very logical and we might have reorganize that structure for IO efficiency.

Yeah agree that some re-org is in order.

ddn0 committed:
build: Pin jupyter client version

The pytest-notebook package depends on a older version of nbconvert
which in turn uses a nbclient package that seems incompatible with the
recent jupyter-client 7.

This leads to errors like:

  TypeError: 'coroutine' object is not subscriptable

ddn0 commented:
Mirror of https://github.com/KatanaGraph/katana-enterprise/pull/1775
